st2common/st2common/models/api/action.py

```diff
@@ -148,6 +149,11 @@ class ActionAPI(BaseAPI):
                 "patternProperties": {
                     "^\w+$": util_schema.get_draft_schema()
                 }
+            },
+            "tags": {
+                "description": "User associated metadata assigned to this object.",
```

Comment:
To me it depends on what we use this for.  If we are going to stick to key value pairs then I much prefer tags.  If we think we will expand on this and have more complicated objects that I believe we should use metadata.  Personally, I like the tag model better.  It may not be as flexible but it is simple.